app/code/Magento/CatalogGraphQl/Model/Resolver/ProductId.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver;
+
+use Magento\CatalogGraphQl\Model\Resolver\Product\DataProvider\ProductDataProviderInterface;
+use Magento\CatalogGraphQl\Model\Resolver\Product\ProductFieldsSelector;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+
+/**
+ * Product by ID resolver, used for GraphQL request processing.
+ */
+class ProductId implements ResolverInterface
+{
+    /**
+     * @var ProductDataProviderInterface
+     */
+    private $productDataProvider;
+
+    /**
+     * @var ProductFieldsSelector
+     */
+    private $productFieldsSelector;
+
+    /**
+     * ProductId constructor.
+     *
+     * @param ProductDataProviderInterface $productDataProvider
+     * @param ProductFieldsSelector $productFieldsSelector
+     */
+    public function __construct(
+        ProductDataProviderInterface $productDataProvider,
+        ProductFieldsSelector $productFieldsSelector
+    ) {
+        $this->productDataProvider = $productDataProvider;
+        $this->productFieldsSelector = $productFieldsSelector;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $productId = (int) ($args['id'] ?? 0);
+        $fields = $this->productFieldsSelector->getProductFieldsFromInfo($info);
+
+        if (!$productId) {
+            throw new GraphQlInputException(
+                __("'id' input argument is required.")
+            );
+        }
+
+        /** @var \Magento\Catalog\Model\Product $product */
+        $product = $this->productDataProvider->getProductById($productId, $fields);
```

Comment:
Have you tried the following scenario?
Pass "-1" as the product ID in the GraphQL query 😎

Reply:
I have not tried this scenario before.
I've just added the additional validation rule for throwing the exception if the ID argument is negative.
<img width="1372" alt="Screenshot 2020-06-09 at 16 15 14" src="https://user-images.githubusercontent.com/31502344/84152012-ca925900-aa6c-11ea-8afb-b2dc7edc540b.png">
Thank you!